server/src/main/java/org/opensearch/snapshots/SnapshotUtils.java

```diff
@@ -135,4 +143,36 @@ public static List<String> filterIndices(List<String> availableIndices, String[]
         }
         return Collections.unmodifiableList(new ArrayList<>(result));
     }
+
+    /**
+     * Validates if there are any remote snapshots backing an index
+     * @param currentState current cluster state
+     * @param snapshotIds list of snapshot Ids to be verified
+     * @param repoName repo name for which the verification is being done
+     */
+    public static void validateSnapshotsBackingAnyIndex(ClusterState currentState, List<SnapshotId> snapshotIds, String repoName) {
+        final Map<String, SnapshotId> uuidToSnapshotId = new HashMap<>();
+        final Set<String> snapshotsToBeNotDeleted = new HashSet<>();
+        ImmutableOpenMap<String, IndexMetadata> indicesMap = currentState.getMetadata().getIndices();
+        snapshotIds.forEach(snapshotId -> uuidToSnapshotId.put(snapshotId.getUUID(), snapshotId));
+
+        for (Iterator<IndexMetadata> it = indicesMap.valuesIt(); it.hasNext();) {
```

Comment:
Minor, but it looks like the prevailing style for iterating over this type is:

```
for (ObjectCursor<IndexMetadata> cursor : state.metadata().indices().values()) {
    IndexMetadata metadata = cursor.value;
}
```

Reply:
Applied the suggestion